### DISCUSSION TITLE
Nitrogen Fixation in CLM6

### DISCUSSION THREAD

**Discussion by wwieder** [NCAR/LMWG_dev#73] [General]
This thread is prompted by simulations in #72, which uses an updated temperature function (below in blue) to calculate C costs of N fix in FUN (second figure, below see also [CTSM draft PR](https://github.com/ESCOMP/CTSM/pull/2869)).  Default temperature function is in orange, and used in #70.

![image](https://github.com/user-attachments/assets/33560b3e-5cc4-4da3-8321-b6282632d837)
 
![image](https://github.com/user-attachments/assets/1104bdd5-dd89-4514-a92b-516da6028cfc)

The Bytnerowicz function (blue lines above) seems to be functioning as expected, with [lower N fix rates](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/set2/set2_ANN_NFIX.gif), [GPP](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/set2/set2_ANN_GPP.gif), [LAI](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/set2/set2_ANN_TLAI.gif) and [ecosystem C](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/set2/set2_ANN_TOTECOSYSC.gif) than a control CLM6 simulation, especially in colder ecosystems (see [diagnostics](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/setsIndex.html) posted by @olyson). 

![image](https://github.com/user-attachments/assets/059926e4-fd70-49cf-94c9-3a5cccdbdd26)

This discussion, however, focuses more on the strong increase in N Fixation rates in our current simulations, compared to older CLM5 runs.  Specifically, baseline N fixation rates are 2x what they were in CLM5 (> 110 vs <55 Tg N /y with the default [CLM6](https://webext.cgd.ucar.edu/I20TR/ctsm530_f19_nfix_hist2/lnd/ctsm530_f19_nfix_hist2_2004_2023-ctsm530_f19_PPE_hist_2004_2023/set5/set5_cn.txt) vs. [CLM5 cases](https://webext.cgd.ucar.edu/I20TR/clm50_release-clm5.0.20_1deg_GSWP3V1_isofix2_hist/lnd/clm50_release-clm5.0.20_1deg_GSWP3V1_isofix2_hist.1995_2014-clm50_release-clm5.0.20_1deg_GSWP3V1_isofix_hist.1995_2014/set5/set5_cn.txt).  Note, this isn't a 100% clean comparison, as CLM6 vs. simulations have different datm, end dates, surface datasets, and more.  

A few things may have caused this increase in Nfix including 
- Fixing the FUN bug in #8, which [increased Nfix 50% in 1850](https://webext.cgd.ucar.edu/I1850/ctsm51d130_ne30pg3_GSWP3V1_LMWG_dev8_1850/lnd/ctsm51d130_ne30pg3_GSWP3V1_LMWG_dev8_1850.2_20-ctsm51_cesm23a16bctsm51d130_ne30pg3ne30pg3mg17_GSWP3V1_1850.2_20/set5/set5_cn.txt) (40 -> 57 Tg N/y)
- Overall higher productivity in CLM6 than CLM5 (GPP ~ 150 vs. 120 Pg C /y), which allows for a higher NPP_NUPTAKE (29 vs. 21 PgC/y), both at the end of the historical period.
- **Are there other parameter changes that could cause these differences?** @linniahawkins 

To make things more confusing: why is [ILAMB](https://webext.cgd.ucar.edu/I20TR/ctsm53n04ctsm52028_f09_hist/lnd/_build_ctsm53n04ctsm52028_f09_hist/EcosystemandCarbonCycle/NitrogenFixation/Davies-Barnard/Davies-Barnard.html) is giving much lower Nfix estimates than our own [diagnostics](https://webext.cgd.ucar.edu/I20TR/ctsm53n04ctsm52028_f09_hist/lnd/ctsm53n04ctsm52028_f09_hist_2004_2023-ctsm52026_f09_hist_2004_2023/set5/set5_cn.txt) for the same run. (40 vs. 120 Tg N /y in #69)?!  It makes me wonder if we're providing a different variable into ILAMB results, @olyson?

**Comment by olyson** [comment]
"FFIX_TO_SMINN" is being provided to ILAMB.
I see 39.1 in ILAMB for #69 and 39.5 for #69  in the diagnostics?

**Comment by olyson** [comment]
Oh, I see you are using NFIX, not FFIX_TO_SMINN.  I was thinking you had suggested FFIX_TO_SMINN for ILAMB, but I can't find proof of that :)

**Reply by wwieder** [reply]
OK, i think we should change that in our ILAMB workflow.  
- FFIX_TO_SMINN is supposed to be freeliving nfix (that would be added to the soil mineral N pool.  
- NFIX is symbiotic N fixation (think soybeans, alder, or mesquite), that's passed directly to the plant.  

I think the observations that are being used for ILAMB are more related to NFIX?

**Comment by olyson** [comment]
The reference paper for the dataset seems to indicate that it is biological nitrogen fixation, which is composed of freeliving and symbiotic nitrogen fixation.  I don't know anything about this obviously but does that mean we should be using FFIX_TO_SMINN+NFIX?   It says "Two primary classes of BNF can be distinguished: symbiotic (also known as associative or nodulating fixation) (Granhall, [1981]) and free-living (also known as nonsymbiotic or asymbiotic) (Reed et al., [2011]).  Free-living biological nitrogen fixation makes up at least a third of the terrestrial total"

**Reply by wwieder** [reply]
Yep.  That makes sense to combine the two fluxes.

**Comment by linniahawkins** [comment]
Our preliminary tuning could have caused these increases in Nfix. I thought we would have reduced Nfix by reducing productivity, but may have increased Nfix by increasing leafC:N depending on whether the region was nitrogen limited (I think).

Parameters we modified: slatop, leafcn, froot_leaf, medlynslope, kmax 
More detail [here](https://docs.google.com/document/d/1hqpvFDNXZhTgXm1gJsfFdkr8pgSxED1nYG2bVXvC2-4/edit?usp=sharing)

**Comment by wwieder** [comment]
One of the positive aspects to the new Nfix temperature function was the reduction in high latitude Nfix, illustrated above. @linniahawkins noted yesterday, however that the default parameter file has `FUN_fracfixers = 1.0`, which is a hold over from our efforts to save dead arctic vegetation in #2.  

There is some concern that hand calibration for CLM5.3 that aimed to tune down high arctic LAI biases pushed arctic shrub and grasses to the upper limit of reasonable values for leafCN and and slatop. One suggestion would be to undo some of these changes: 
- setting FUN_fracfixers back to 0.25 and 
- reverting leafCN and slatop to their previous values.

Defining those previous values for leafCN and slatop is a little tricky for these pft's as they were changed from CLM5.0 -> 5.1 (with arctic phenology changes) and again with modifications to try and save the dead arctic veg.  We tried some of this this in #50, with lots of results [posted here](https://github.com/NCAR/LMWG_dev/discussions/3#discussioncomment-8055964).  I'll note here that the focus on the dead veg series: 
- Focused on fracfixers and froot_leaf, it looks like we didn't modify leafcn or slatop in [# 34](https://github.com/NCAR/LMWG_dev/issues/34), or subsequent series of ABCDE runs
- Largely used GSWP3 forcing, there are only a few runs that have CRU-JRA.

Seems like we need some additional simulations to evaluate survivability with a modern code base, less arctic N fixation, and more reasonable values for leaf traits.

**Comment by linniahawkins** [comment]
As a first order test, I am running an initial sparse grid simulation with the following setup:

codebase: ctsm5.3.012 + new Nfix temperature function (https://github.com/wwieder/ctsm/tree/BNF_v2)
parameter changes: 
- FUN_fracfixers = 0.25 for arctic grass and shrubs
- leafcn, slatop, froot_leaf reverted to CLM5.1 default values for arctic grass and shrubs

I simultaneously tried to bring back C4 grass with the following parameter changes:
- Set froot_leaf back to default for BL decid. tropical tree and C4 grass (undoing Fang’s parameter changes). 
- relax nutrient costs for C4 grass only (akc_active, akn_active, ekc_active, ekn_active, kc_nonmyc, kn_nonmyc parameters to minimum)

finidat = '/glade/derecho/scratch/wwieder/archive/ctsm530_f19_nfix_pSASU/rest/0161-01-01-00000/ctsm530_f19_nfix_pSASU.clm2.r.0161-01-01-00000.nc'
paramfile = '/glade/work/linnia/CLM6-PPE/ctsm53012_Nfix_tests/paramfiles/Nfix0001.nc'

simulation protocol: 40y AD + 120y SASU + 40y postSASU + transient 1850-2023

currently @ 1975, will post results shortly.

**Reply by wwieder** [reply]
Thanks Linnia.  I'm curious to see how they look.

**Reply by linniahawkins** [reply]
Results:

- survivability checks passed (no reduction in number of gridcells with LAI>0.1 for any PFT on sparsegrid)
- C4 grass expanded (increase in the number of gridcells with C4 LAI>0.1) 
- Clear increase in C4 grass LAI (figure below). 
- Significant reduction in arctic LAI (larger than I expected). 


![Screenshot 2024-11-19 at 2 04 43 PM](https://github.com/user-attachments/assets/cd438651-c362-46b8-b617-1c206d2f5904)

**Reply by linniahawkins** [reply]
I do not have confidence that this parameter set won't cause survivability issues in the arctic in full grid simulations. 

option 1: try it as is on fullgrid
option 2: try to do a little more hand tuning on sparse grid first
option 3: wait for the first LHC PPE ensemble and use emulators to develop a new preliminary tuning.

**Reply by wwieder** [reply]
I'm somewhat inclined to try a full grid spinup and optimistic that we'll be able to assess survivability after the AD run?  I'll try this tonight.

**Reply by wwieder** [reply]
just looking at AD spinup from your test case on the left with the 'default' CTSM5.3 parameters on the right, both using the new Nfix scheme.  Arctic survival, inferred here from TOTVEGC looks slightly better across part of N Canada. C stocks also are higher in the Sahel and even W Australia, suggesting the C4 grasses are doing OK too.  Will continue with SASU and pSASU spinup on this case (and include as a new issue) 
<img width="1164" alt="image" src="https://github.com/user-attachments/assets/15fcdaeb-6212-4194-a216-5f094d98e226">

**Reply by linniahawkins** [reply]
That's encouraging!
